deltacat/compute/compactor/utils/io.py

```diff
@@ -207,3 +192,25 @@ def limit_input_deltas(
     logger.info(f"Input uniform delta count: {len(rebatched_da_list)}")
 
     return rebatched_da_list, hash_bucket_count, latest_stream_position
+
+
+def getLastCompactedDeltaStreamPosition(
+        source_partition_locator: PartitionLocator,
+        deltacat_storage=unimplemented_deltacat_storage) -> int:
+    stream_locator = source_partition_locator.stream_locator
+    namespace = stream_locator.namespace
+    table_name = stream_locator.table_name
+    table_version = stream_locator.table_version
+    partition_values = source_partition_locator.partition_values
+    end_position_inclusive = deltacat_storage.get_partition(stream_locator, partition_values).stream_position
+    deltas_list_result = deltacat_storage.list_deltas(
+        namespace,
+        table_name,
+        partition_values,
+        table_version,
+        None,
+        end_position_inclusive,
+        True,
+    )
+    deltas_list_result = deltas_list_result.all_items()
+    return int(deltas_list_result[-1]['properties']['parent_stream_position'])
```

Comment:
This implementation is specific to amazon internal catalog. Catalogs like IceBerg wouldn't have this field and will be enforced to have it as properties if they onboard. So, I would have this code within driver and compact_partition take it as arg.

Reply:
The other way to handle this is to just leave it as a storage interface that each catalog can provide their own implementation for.

The general implementation for any catalog managed by DeltaCAT could also be provided, which always reads the high watermark from the last round completion file or returns `None` if no round completion file exists (i.e. even this implementation is only useful for migration scenarios).

Reply:
Sounds good. removed the code. Will rely on driver side for inferring rebase partition's high watermark.